app/kumactl/cmd/delete/delete_meshes_test.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/Kong/kuma/app/kumactl/cmd"
+	config_proto "github.com/Kong/kuma/pkg/config/app/kumactl/v1alpha1"
+	"github.com/Kong/kuma/pkg/core/resources/apis/mesh"
+	core_model "github.com/Kong/kuma/pkg/core/resources/model"
+	core_store "github.com/Kong/kuma/pkg/core/resources/store"
+	memory_resources "github.com/Kong/kuma/pkg/plugins/resources/memory"
+	test_model "github.com/Kong/kuma/pkg/test/resources/model"
+
+	kumactl_cmd "github.com/Kong/kuma/app/kumactl/pkg/cmd"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+)
+
+var _ = Describe("kumactl delete mesh", func() {
+
+	sampleMeshes := []*mesh.MeshResource{
+		{
+			Spec: v1alpha1.Mesh{
+				Mtls: &v1alpha1.Mesh_Mtls{
+					Enabled: true,
+					Ca: &v1alpha1.CertificateAuthority{
+						Type: &v1alpha1.CertificateAuthority_Builtin_{
+							Builtin: &v1alpha1.CertificateAuthority_Builtin{},
+						},
+					},
+				},
+				Logging: &v1alpha1.Logging{
+					AccessLogs: &v1alpha1.Logging_AccessLogs{
+						Enabled:  true,
+						FilePath: "/tmp/access.log",
+					},
+				},
+			},
+			Meta: &test_model.ResourceMeta{
+				Mesh:      "mesh1",
+				Name:      "mesh1",
+				Namespace: "default",
+			},
+		},
+		{
+			Spec: v1alpha1.Mesh{
+				Mtls: &v1alpha1.Mesh_Mtls{
+					Enabled: false,
+					Ca: &v1alpha1.CertificateAuthority{
+						Type: &v1alpha1.CertificateAuthority_Builtin_{
+							Builtin: &v1alpha1.CertificateAuthority_Builtin{},
+						},
+					},
+				},
+			},
+			Meta: &test_model.ResourceMeta{
+				Mesh:      "mesh2",
+				Name:      "mesh2",
+				Namespace: "default",
+			},
+		},
+	}
+
+	Describe("DeleteMeshCommand", func() {
+
+		var rootCtx *kumactl_cmd.RootContext
+		var rootCmd *cobra.Command
+		var buf *bytes.Buffer
+		var store core_store.ResourceStore
+
+		BeforeEach(func() {
+			// setup
+			rootCtx = &kumactl_cmd.RootContext{
+				Runtime: kumactl_cmd.RootRuntime{
+					Now: func() time.Time { return time.Now() },
+					NewResourceStore: func(*config_proto.ControlPlaneCoordinates_ApiServer) (core_store.ResourceStore, error) {
+						return store, nil
+					},
+				},
+			}
+
+			store = memory_resources.NewStore()
+
+			for _, ds := range sampleMeshes {
+				key := core_model.ResourceKey{
+					Mesh:      ds.Meta.GetMesh(),
+					Namespace: ds.Meta.GetNamespace(),
+					Name:      ds.Meta.GetName(),
+				}
+				err := store.Create(context.Background(), ds, core_store.CreateBy(key))
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			rootCmd = cmd.NewRootCmd(rootCtx)
+			buf = &bytes.Buffer{}
+			rootCmd.SetOut(buf)
+		})
+
+		It("should throw an error in case of no args", func() {
+			// given
+			rootCmd.SetArgs([]string{
+				"--config-file", filepath.Join("..", "testdata", "sample-kumactl.config.yaml"),
+				"delete", "mesh"})
+
+			// when
+			err := rootCmd.Execute()
+
+			// then
+			Expect(err).To(HaveOccurred())
+			// and
+			Expect(err.Error()).To(Equal("accepts 1 arg(s), received 0"))
+		})
```

Comment:
Let's also check the output generated by the command (to guarantee proper behaviour and clear message from a user perspective).

See https://github.com/Kong/kuma/blob/8bb10ce/app/kumactl/cmd/config/config_control_planes_add_test.go#L59-L67 for example